homeassistant/components/anova/config_flow.py

```diff
@@ -0,0 +1,57 @@
+"""Config flow for Anova."""
+from __future__ import annotations
+
+from anova_wifi import AnovaApi, AnovaOffline, InvalidLogin, NoDevicesFound
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN
+
+
+class AnovaConfligFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Sets up a config flow for Anova."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            api = AnovaApi(
+                aiohttp_client.async_get_clientsession(self.hass),
+                user_input[CONF_USERNAME],
+                user_input[CONF_PASSWORD],
+            )
+            await self.async_set_unique_id(user_input[CONF_USERNAME].lower())
+            self._abort_if_unique_id_configured()
+            try:
+                await api.authenticate()
+                devices = await api.get_devices()
+            except AnovaOffline:
+                errors["base"] = "cannot_connect"
+            except InvalidLogin:
+                errors["base"] = "invalid_auth"
+            except NoDevicesFound:
+                errors["base"] = "no_devices_found"
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+            else:
+                device_list = [(device.device_key, device.type) for device in devices]
+                return self.async_create_entry(
+                    title="Anova",
+                    data={"jwt": api.jwt, "devices": device_list},
```

Comment:
Do we need to store the device list here?

Reply:
I think it would be better to get the list at setup time so no devices don't require deleting and re-adding the entry

Reply:
Potentially not - it would probably be better to get the device list in init - as that way every time the integration starts up it will find new devices it did not previously know about. I'll make that change

Reply:
I am partially remembering why I did not originally. The anova api finds no devices if the device is offline, meaning I should raise configentrynotready  if I am doing it in init. 

Perhaps the best approach is to store device information in config flow, then in init check for new devices, and if there are any new devices, update the config entry.

That way if I have two devices, one of which is offline and one of which is online, I'll set up the one that is online when home assistant starts up, but because I have the offline device information, I can still set it up, it will just be unavailable, until I turn it on and then it will become available in HA, instead of me having to restart in order for it to show up

Thoughts?

Reply:
Sounds good

Reply:
Please add comments about why it's going to be implemented that way, so we don't have to guess later when we look at this years from now